helper/resource/testing.go

```diff
@@ -752,10 +755,47 @@ func ComposeAggregateTestCheckFunc(fs ...TestCheckFunc) TestCheckFunc {
 	}
 }
 
-// TestCheckResourceAttrSet is a TestCheckFunc which ensures a value
-// exists in state for the given name/key combination. It is useful when
-// testing that computed values were set, when it is not possible to
-// know ahead of time what the values will be.
+// TestCheckResourceAttrSet ensures any value exists in the state for the
+// given name and key combination. The opposite of this TestCheckFunc is
+// TestCheckNoResourceAttr. State value checking is only recommended for
+// testing Computed attributes and attribute defaults.
+//
+// Use this as a last resort when a more specific TestCheckFunc cannot be
+// implemented, such as:
+//
+// - TestCheckResourceAttr: Equality checking of non-TypeSet state value.
+// - TestCheckResourceAttrPair: Equality checking of non-TypeSet state value,
+//                              based on another state value.
+// - TestCheckTypeSet*: Equality checking of TypeSet state values.
+// - TestMatchResourceAttr: Regular expression checking of non-TypeSet state
+//                          value.
+// - TestMatchTypeSet*: Regular expression checking on TypeSet state values.
+//
+// For managed resources, the name parameter is combination of the resource
+// type, a period (.), and the name label. The name for the below example
+// configuration would be "myprovider_thing.example".
+//
+//     resource "myprovider_thing" "example" { ... }
+//
+// For data sources, the name parameter is a combination of the keyword "data",
+// a period (.), the data source type, a period (.), and the name label. The
+// name for the below example configuration would be
+// "data.myprovider_thing.example".
+//
+//     data "myprovider_thing" "example" { ... }
+//
+// The key parameter is an attribute path in Terraform CLI 0.11 and earlier
+// "flatmap" syntax. Keys start with the attribute name of a top-level
+// attribute. Use the following special key syntax to inspect underlying
+// values of a list or map attribute:
+//
+// - .{NUMBER}: List value at index, e.g. .0 to inspect the first element
+// - .{KEY}: Map value at key, e.g. .example to inspect the example key value
```

Comment:
Yeah totally -- I meant to implement [Go testable examples](https://go.dev/blog/examples) for these as part of this update.

It's going to be a little awkward though, because its only recommended to implement these for Computed values, but I'll essentially have to show the configuration equivalent so it makes more sense looking at it.

Reply:
I've uploaded some TestCheckFunc examples.